packages/dashboard/src/components/alert-manager.tsx

```diff
@@ -294,44 +288,42 @@ export const AlertStore = React.memo(() => {
         );
       } catch (e) {
         console.log(
-          `Alert response could not be found for ${alertRequest.id}, ${(e as Error).message}`,
+          `Error retrieving alert response of ID ${alertRequest.id}, ${(e as Error).message}`,
         );
         setOpenAlerts((prev) => {
           return {
             ...prev,
             [alertRequest.id]: alertRequest,
           };
         });
-        AppEvents.refreshAlert.next();
         return;
       }
     };
 
     const subs: Subscription[] = [];
 
     subs.push(
-      rmf.alertRequestsObsStore.subscribe(
-        async (alertRequest) => await pushAlertsToBeDisplayed(alertRequest),
+      rmf.alertRequestsObsStore.subscribe((alertRequest) =>
+        AppEvents.alertListOpenedAlert.next(alertRequest),
```

Comment:
We should filter first before emitting onto `alertListOpenedAlert`, doing it after results alerts in `alertListOpenedAlert` not actually being opened.

Hypothetically, suppose we want to collect some analytics on how long does it take for an alert to be responded or dismissed, if we do the filtering before, we can look at metrics on `alertListOpenedAlert`.

Reply:
https://github.com/open-rmf/rmf-web/pull/948/commits/6494c01c211186b803419591f93ac8fbe3eef18a